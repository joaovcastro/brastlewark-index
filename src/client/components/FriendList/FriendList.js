import React from 'react';
import { Link } from 'react-router-dom';
import {
  Grid,
  CardMedia,
  Typography,
  CardContent,
  Divider,
} from '@material-ui/core';
import { array } from 'prop-types';

const FriendList = ({ list }) => (
  <CardContent>
    <Grid container spacing={8} justify="space-between">
      <Grid item xs={12} style={{ marginBottom: '15px' }}>
        <div style={{ Width: '100%' }}>
          <Typography gutterBottom variant="h5" style={{ color: ' #3a5637' }}>
            Friends
          </Typography>
        </div>
        <Divider />
      </Grid>
      {list.map(
        friend =>
          friend && (
            <Grid item key={friend.name} sm={3} style={{ textAlign: 'center' }}>
              <Link
                key={friend.id}
                to={{
                  pathname: `/gnomes/${friend.id}`,
                }}
                onClick={() => window.location.refresh()}
              >
                <CardMedia
                  image={friend.thumbnail}
                  title="Avatar"
                  style={{
                    height: '150px',
                    width: '150px',
                    borderRadius: '100%',
                    marginLeft: 'calc(50% - 75px)',
                  }}
                />
                <Typography
                  gutterBottom
                  variant="h6"
                  style={{ paddingTop: '10px' }}
                >
                  {friend.name}
                </Typography>
              </Link>
            </Grid>
          ),
      )}
    </Grid>
  </CardContent>
);

FriendList.propTypes = {
  list: array.isRequired,
};

export default FriendList;
