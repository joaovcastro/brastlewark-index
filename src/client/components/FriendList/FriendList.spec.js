import React from 'react';
import { shallow } from 'enzyme';

import FriendList from './FriendList';

describe('<FriendList />', () => {
  it('correctly renders', () => {
    const wrapper = shallow(<FriendList list={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
