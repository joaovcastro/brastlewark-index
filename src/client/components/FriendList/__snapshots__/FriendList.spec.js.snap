// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FriendList /> correctly renders 1`] = `ShallowWrapper {}`;
